packages/gatsby-plugin-typescript/src/gatsby-node.js

```diff
@@ -16,6 +16,42 @@ function onCreateBabelConfig({ actions }, options) {
   })
 }
 
+function onCreateWebpackConfig({ actions, loaders }) {
+  if (typeof loaders?.js !== `function`) {
+    return
+  }
+
+  let doesUsedGatsbyVersionSupportResourceQuery
+  try {
+    const { version } = require(`gatsby/package.json`)
+    const [major, minor] = version.split(`.`).map(Number)
+    doesUsedGatsbyVersionSupportResourceQuery =
+      (major === 4 && minor >= 19) || major > 4
+  } catch {
+    doesUsedGatsbyVersionSupportResourceQuery = true
+  }
```

Comment:
should we get rid of that and assume that those versions of plugins will run exclussively with gatsby@5? (not necasairly in this PR for clearer paper trail as that wouldn't be just revert anymore)

Reply:
created story 57312